torch/distributed/fsdp/_init_utils.py

```diff
@@ -516,6 +516,10 @@ def _init_param_handle_from_module(
         _sync_module_params_and_buffers(
             fully_sharded_module, managed_params, state.process_group
         )
+        if hasattr(state, '_inter_node_pg'):
```

Comment:
nit: I wonder if this might be brittle for checking if we are using HSDP. Perhaps, we can do `getattr(state, "_inter_node_pg", None) is not None`?

Reply:
yes, I have changed this